app/utils/External.ts

```diff
@@ -32,6 +31,22 @@ function getContentType(type?: string) {
     return { 'content-type': type || 'application/json' };
 }
 
+// NOTE: Regex taken from https://stackoverflow.com/questions/23054475/javascript-regex-for-extracting-filename-from-content-disposition-header
+const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
+function getFilenameFromHeaders(headers: Headers, fallbackFilename: string) {
+    let filename = fallbackFilename;
+    const contentDispositionHeader = headers.get('content-disposition');
+
+    if (contentDispositionHeader && contentDispositionHeader?.indexOf('attachment') >= 0) {
+        const matches = filenameRegex.exec(contentDispositionHeader);
+        if (matches?.[1]) {
+            filename = matches[1].replace(/['"]/g, '');
+        }
+    }
+
+    return filename;
+}
```

Comment:
WDYT about removing the let `filename` variable in order of having a `return` statements?

It may make the code more predictable and understandable 💪 

```suggestion
function getFilenameFromHeaders(headers: Headers, fallbackFilename: string) {
    const contentDispositionHeader = headers.get('content-disposition');

    if (contentDispositionHeader && contentDispositionHeader?.indexOf('attachment') >= 0) {
        const matches = filenameRegex.exec(contentDispositionHeader);
        if (matches?.[1]) {
            const filename = matches[1].replace(/['"]/g, '');
            return filename;
        }
    }

    return fallbackFilename;
}
```

Reply:
OK, done.